feat: add scheduled GitHub Action to regenerate provider registry

TylerBarnes opened the pull request:
Adds a scheduled workflow that automatically regenerates the provider registry every 6 hours, keeping the model/provider data fresh without manual intervention.

The workflow fetches the latest data from models.dev and Netlify, generates the registry files, formats them with Prettier, and commits directly to main when changes are detected. It also creates a changeset for @mastra/core to trigger a patch version bump.

This works alongside the existing runtime auto-refresh system - the GitHub Action keeps the repo source files up-to-date while the runtime system keeps local dev environments fresh between git pulls.

Can be manually triggered from the Actions tab for testing.

TylerBarnes committed:
feat: add scheduled workflow to regenerate provider registry

- Runs daily at midnight UTC
- Automatically generates provider registry files
- Formats with Prettier to prevent lint errors
- Creates changeset for @mastra/core patch bump
- Commits directly to main when changes detected
- Silent and transparent with [skip ci] flag
- Supports manual triggering via workflow_dispatch

TylerBarnes committed:
chore: update provider registry schedule to every 6 hours

Aligns with runtime auto-refresh cadence for better consistency

TylerBarnes committed:
fix: improve changeset filename uniqueness and git add specificity

- Add timestamp to changeset filename to prevent conflicts on same day
- Use specific changeset file variable instead of wildcard in git add
- Export CHANGESET_FILE to env for use in subsequent steps

Addresses: https://github.com/mastra-ai/mastra/pull/8834#issuecomment-3403180968

TylerBarnes committed:
fix: format workflow file with Prettier

wardpeet reviewed (approved):
LGTM, only need to change the user to use the dane bot

TylerBarnes committed:
feat: use Dane bot for provider registry commits

Use Dane GitHub App bot instead of github-actions bot to ensure
subsequent GitHub Actions are triggered correctly.

Implements Ward's feedback:
https://github.com/mastra-ai/mastra/pull/8834#discussion_r1875123456

TylerBarnes commented on .github/workflows/regenerate-provider-registry.yml:
```diff
@@ -0,0 +1,95 @@
+    if: ${{ github.repository == 'mastra-ai/mastra' }}
+    runs-on: ubuntu-latest
+
+    steps:
```
Thanks, updated it in https://github.com/mastra-ai/mastra/pull/8834/commits/565de8b3fc57a7e33ef53ea263a2cd30fe9878f0

TylerBarnes committed:
fix: ensure directories exist before writing registry files

Add path import and mkdir calls to create necessary directories
before writing provider-registry.json and provider-types.generated.d.ts.

This is needed because:
1. The dist/ directory is gitignored, so it won't exist in CI or fresh checkouts
2. Both the generation script and runtime syncGateways() use writeRegistryFiles()
3. Without mkdir, writing to dist/llm/model/ fails when the directory doesn't exist
